include/json/value.h

```diff
@@ -585,6 +585,22 @@ class JSON_API Value {
   iterator begin();
   iterator end();
 
+  /// \brief Returns a reference to the first element in the `Value`.
+  /// Requires that this value holds an array or json object, with at least one element.
+  const Value& front() const;
+
+  /// \brief Returns a reference to the last element in the `Value`.
+  /// Requires that value holds an array or json object, with at least one element.
+  const Value& back() const;
+
+  /// \brief Returns a reference to the first element in the `Value`.
+  /// Requires that this value holds an array or json object, with at least one element.
+  Value& front();
```

Comment:
Let's have the order here match the order in the .cpp file, which I think is the better ordering:

```c++
    const Value& front() const;
    Value& front();
    const Value& back() const;
    Value& back();
```